ad-joining/register-computer/join.ps1

```diff
@@ -27,13 +27,61 @@
 #--------------------------------------------------------------------------------------------
 
 $ErrorActionPreference = "Stop"
+$InformationPreference = "Continue";
+
+$metadataUri = "http://metadata.google.internal/computeMetadata/v1/instance";
+$metadataHeaders = @{"Metadata-Flavor" = "Google"};
+
+$enableDiagnostics = $false;
+try
+{
+    $enableDiagnostics = (Invoke-RestMethod -Headers $metadataHeaders `
+        -Uri "$($metadataUri)/attributes/enable-adjoin-diagnostics");
+
+    if($enableDiagnostics)
+    {
+        Write-Information -MessageData "AD Join diagnostics: Enabled"; 
+
+        $diagnosticsBucket = (Invoke-RestMethod -Headers $metadataHeaders `
+            -Uri "$($metadataUri)/attributes/adjoin-diagnostics-bucket");
+
+        if($diagnosticsBucket -eq "")
+        {
+            throw New-Object System.ArgumentException "AD Join diagnostics enabled but bucket not set. Point adjoin-diagnostics-bucket metadata to a GCS bucket the service account has write access to.";
+        }
+
+        $diagnosticCaptureFile = "$env:SystemRoot\temp\capture.etl";
```

Comment:
why not ${env:TEMP}?

Reply:
$env:TEMP resolves to something else when the system boots into a user session so I have been using this way to resolve to the same directory. In this case this is not relevant as we are capturing during sysprep and write the capture off to GCS. Will change to your suggestion.